### PR TITLE
[fix #3913] do not logout when changing mailserver

### LINKED ISSUE #3913: Fix mailserver configuration change
When the mailserver configuration is changed:
- make sure that  the previous mailserver is disconnected from peers (will involve json_rpc API)
- the new one is added (should already be done by calling initialize-offline-inbox)
- do not logout  when saving mailserver config

### PR DESCRIPTION
**Pull request by yenda**
fixes #3913 

### Summary:

- do not logout and remove previous mailserver
from peers when changing mailserver
- rename wnode mailserver
- move transport.inbox to mailserver.core
- fix all subs and db keys

#### Platforms
- Android
- iOS
- macOS
- Linux

#### Areas that maybe impacted
**Functional**
- mailservers
- fleet

### Steps to test:
- Select a mailserver in an account before upgrading
- After upgrade, same mailserver should still be selected
- Change selected mailserver, it should not logout and previous mailserver should disconnect (remove-peer success true in the logs).

- Selected mailserver should persist when restarting the app
- Adding custom mailserver should still work and be persisted accross restarts
- Changing fleet should still work

status: ready

### CODE REVIEW

File: src/status_im/mailserver/db.cljs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+(ns status-im.mailserver.db
+  (:require [status-im.mailserver.core :as mailserver]
+            [cljs.spec.alpha :as spec]))
+
+(spec/def :mailserver/state (spec/nilable #{:disconnected :connecting :added :connected :error}))
+(spec/def :mailserver/current-id (spec/nilable keyword?))
+
+(spec/def ::not-blank-string (spec/and string? seq))
```
Comment (rasom) [inline src/status_im/mailserver/db.cljs]: `:global/not-empty-string` is used below, is it different?

File: src/status_im/mailserver/db.cljs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+(spec/def :mailserver/generating-sym-key? boolean?)
+(spec/def :mailserver/mailserver (spec/keys :req-un [:mailserver/address :mailserver/name :mailserver/id]
+                                            :opt-un [:mailserver/sym-key-id
+                                                     :mailserver/generating-sym-key?
+                                                     :mailserver/user-defined
+                                                     :mailserver/password]))
+
+(spec/def :mailserver/mailservers (spec/nilable (spec/map-of keyword? (spec/map-of :mailserver/id :mailserver/mailserver))))
+
+(spec/def :request/from pos-int?)
```
Comment (rasom) [inline src/status_im/mailserver/db.cljs:22]: can we be more specific here and check the order of magnitude for instance (assuming that's future dates), or it doesn't make sense?

File: src/status_im/mailserver/subs.cljs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+(re-frame/reg-sub
+ :mailserver.edit/mailserver
+ (fn [db]
+   (get db :mailserver.edit/mailserver)))
+
+(re-frame/reg-sub
+ :mailserver.edit/connected?
+ :<- [:mailserver.edit/mailserver
+      :mailserver/current-id]
```
Comment (janherich) [inline src/status_im/mailserver/subs.cljs]: I believe this is a typo and should be:
```clj
:<- [:mailserver.edit/mailserver]
:<- [:mailserver/current-id]
```
Instead.

**Review by janherich — CHANGES_REQUESTED** [review]

File: src/status_im/mailserver/db.cljs (hunk 2 above)
Comment (yenda) [reply src/status_im/mailserver/db.cljs:22]: could you elaborate I don't understand the idea

**Review by janherich — APPROVED** [review]

File: src/status_im/mailserver/db.cljs (hunk 2 above)
Comment (rasom) [reply src/status_im/mailserver/db.cljs:22]: well, i actually figured that this is not a new code, it's just moved to another file, so nevermind. What i meant is that we can define something like `:global/timestamp` and actually validate that this is a timestamp, not just `pos-int?`, or at least have an alias for this type which would make the spec more detailed. That's it. But probably not the scope of this PR.

**Review by rasom — APPROVED** [review]

**Review by vitvly — APPROVED** [review]

**Comment by yenda** [comment]
@SlizkovaGanna why did you move it back to contributor ?

**Comment by asemiankevich** [comment]
@yenda not sure it is related to this PR but to doublecheck - why `add` button is disabled when adding URL manually? Only qr scan should work? 

image

**Comment by asemiankevich** [comment]
@yenda nevermind i solved it, should be `enode://531e252ec966b7e83f5538c19bf1cde7381cc7949026a6e499b6e998e695751aadf26d4c98d5a4eabfb7cefd31c3c88d600a775f14ed5781520a88ecd25da3c6:status-offline-inbox@35.225.227.79:30504` format

**Comment by asemiankevich** [comment]
@yenda Error when scanning mailserver QR

image


image

**Comment by asemiankevich** [comment]
@yenda all good except same error when i delete mailserver that i just created, would u mind to check that? Thanks

image

**Comment by asemiankevich** [comment]
also 

image

and 

image

**Comment by asemiankevich** [comment]
@yenda 

autotests failed with the following errors:

```
11-06 06:26:23.634  4419  4462 E ReactNativeJS: Error: No protocol method IDeref.-deref defined for type undefined: 
11-06 06:26:23.634  4419  4462 E ReactNativeJS: 
11-06 06:26:23.634  4419  4462 E ReactNativeJS: This error is located at:
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in send-button-view
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in reagent175
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in input-container
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in reagent155
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in reagent141
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTScrollView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in ScrollView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in chat-root
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in reagent138
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in chat
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in reagent137
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in reagent136
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in reagent16
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in n
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in n
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in n
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in n
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in r
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in n
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in n
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in n
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in n
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in n
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in r
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in n
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in n
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in n
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in reagent2
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in root
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in reagent1
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in RCTView
11-06 06:26:23.634  4419  4462 E ReactNativeJS:     in t
```

```
11-06 06:32:27.034  4412  4452 E ReactNativeJS: Error: No protocol method IDeref.-deref defined for type undefined: 
11-06 06:32:27.034  4412  4452 E ReactNativeJS: 
11-06 06:32:27.034  4412  4452 E ReactNativeJS: This error is located at:
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in send-button-view
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in reagent193
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in input-container
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in reagent173
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in reagent160
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTScrollView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in ScrollView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in chat-root
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in reagent157
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in chat
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in reagent156
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in reagent155
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in reagent16
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in n
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in n
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in n
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in n
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in r
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in n
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in n
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in n
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in n
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in n
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in r
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in n
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in n
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in n
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in reagent2
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in root
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in reagent1
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in RCTView
11-06 06:32:27.034  4412  4452 E ReactNativeJS:     in t
```

also there are more , please check out the e2e results above. It does not happen in nightly.

**Comment by asemiankevich** [comment]
- all e2e fails here are not related to this PR. 
- all mentioned issues are fixed
- no regressions observed

**Pull request merged**